From af82f3620a4b860e2c93f228b12fe7d44708a6dc Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Tue, 27 Mar 2018 03:16:55 +0200
Subject: [PATCH 4/4] mach-msm: Fix dependencies for TIMA configs

* Don't force TIMA_LKMAUTH for modules
* Make all TIMA options depend on TIMA

Change-Id: I8c2c81ba8df5ca6a92ab0fbcba2501c6680a4c57
---
 arch/arm/mach-msm/Kconfig | 3 +++
 init/Kconfig              | 1 -
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-msm/Kconfig b/arch/arm/mach-msm/Kconfig
index dfda8d7..6b01a5a 100644
--- a/arch/arm/mach-msm/Kconfig
+++ b/arch/arm/mach-msm/Kconfig
@@ -4775,6 +4775,7 @@ config SWITCH_DUAL_MODEM
 
 config TIMA_RKP
        bool "Enable RKP (Realtime Kernel Protection) TIMA feature"
+       depends on TIMA
        default n
 
 config TIMA_RKP_30
@@ -4845,6 +4846,7 @@ config TIMA
 
 config TIMA_LKMAUTH
        bool "Enable LKM authentication TIMA feature"
+       depends on TIMA && MODULES
        default n
        help
          TIMA authenticates loaded kernel modules. Disable it if you don't
@@ -4860,6 +4862,7 @@ config TIMA_LKMAUTH_CODE_PROT
 
 config TIMA_TEST_INFRA
        tristate "Testing infrastructure for TIMA"
+       depends on TIMA
        default n
        help
          Kernel bits of the TIMA test infrastructure
diff --git a/init/Kconfig b/init/Kconfig
index 192589a..bdb57a5 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -1379,7 +1379,6 @@ config BASE_SMALL
 
 menuconfig MODULES
 	bool "Enable loadable module support"
-	select TIMA_LKMAUTH
 	help
 	  Kernel modules are small pieces of compiled code which can
 	  be inserted in the running kernel, rather than being
-- 
2.7.4

